core-rust/state-manager/src/staging/result.rs

```diff
@@ -180,24 +180,34 @@ impl ProcessedCommitResult {
         );
         let state_hash_tree_diff =
             Self::compute_state_tree_update(store, parent_state_version, &database_updates);
-        let transaction_tree_diff = Self::compute_accu_tree_update::<S, TransactionTreeHash>(
-            store,
-            epoch_transaction_identifiers.state_version,
-            epoch_transaction_identifiers.transaction_hash,
-            parent_state_version,
-            TransactionTreeHash::from(ledger_transaction_hash.into_hash()),
-        );
+
+        let epoch_accu_trees =
+            EpochAwareAccuTreeFactory::new(epoch_identifiers.state_version, parent_state_version);
+
+        let transaction_tree_diff = {
+            let mut collector = CollectingAccuTreeStore::new(store);
+            epoch_accu_trees
+                .create_builder(epoch_identifiers.transaction_hash, &mut collector)
+                .append(TransactionTreeHash::from(
+                    ledger_transaction_hash.into_hash(),
```

Comment:
I feel like we should create a `From<LedgerTransactionHash> for TransactionTreeHash` impl